src/app/core/admin/building-blocks/admin-tabs/admin-tabs.component.ts

```diff
@@ -76,4 +78,22 @@ export class AdminTabsComponent<
       this.tabGroup.focusTab(newTabIndex);
     });
   }
+  getAllTabs(index) {
+    var allTabs = [];
+    for (var i = 0; i < this.tabs?.length; i++) {
+      if (i != index) {
+        allTabs.push("tabs-" + i);
+      }
+    }
+
+    return allTabs;
+  }
+
+  drop(event: CdkDragDrop<string[]>) {
+    const previousIndex = parseInt(event.previousContainer.id.replace("tabs-", ""), 10);
+    const currentIndex = parseInt(event.container.id.replace("tabs-", ""), 10);
```

Comment:
I think you do not actually need this workaround using ids. The drop event also gives you the previous and current index directly, for example see https://github.com/Aam-Digital/ndb-core/blob/9303825461f4b4bf350e05ea4327a2142dc854c6/src/app/core/admin/admin-entity-details/admin-entity-form/admin-entity-form.component.ts#L142-L151

Reply:
Hi @sleidig when I tried this `event.previousContainer.data` and event.container.data` I am not getting the previous and current container data so that's why I am using the `ids` approach for mat tabs.

Reply:
A pity this needs such a workaround but I couldn't find any simpler way either. Good solution then!